core/vm/lfvm/instructions.go

```diff
@@ -306,10 +306,9 @@ func opCallDataCopy(c *context) {
 	}
 
 	c.memory.EnsureCapacity(memOffset64, length64, c)
-	if c.status != RUNNING {
-		return
+	if c.status == RUNNING {
+		c.memory.Set(memOffset64, length64, getData(c.data, dataOffset64, length64))
```

Comment:
I don't understand this change ... is there a difference? If so, it should be commented on to avoid undoing this. Otherwise, handling error cases in the if seems to be the idiomatic style.

Reply:
Did refactor of EnsureCapacity function to return error

Reply:
looks god :+1: